### DISCUSSION TITLE
GitHub usage ideas

### DISCUSSION THREAD

**Discussion by prtda** [SWFSC/glider-lab#27] [Ideas]
_Starting a discussion to list out some possible ways we might try experimenting with github issues and project features..._
***
Since we're all sort of learning this system together, I thought it might be good to have a place to brainstorm some ways to make the most of GitHub. 

Though, these ideas may be more cumbersome than helpful, so of course take it or leave it as needed. Also, caveat, I'm new here :)

#### Project
- Column for location of task, especially for lab work. Right now we have a `Location` column, maybe we should change to `Lab` or something like that. Drop-down options: `La Jolla`, `Humboldt`, `n/a`
- Make a project view for supervisors, based on what they want to see. Make it as easy as possible to check progress for someone who is not necessarily immersed in the day-to-day.
- In general, I think we should all play with "project views" in this experimental phase. 
- When a glider is deployed, should we create an "issue" for that deployment? Assign pilots, ask for help, discuss, etc.? Then we could have a "roadmap" project view of our glider deployments. 

#### Repository
- Have a `deployment` folder in the repo with subfolders that contain the piloting files for a particular deployment. Pilots would "pull" these to their local machines to use in SFMC and push back to the repo when making any changes after sending to the glider. _Maybe this is overkill, as SFMC archives everything already. But it was something I wanted to do on my personal repo when piloting, because I use different computers depending on where I am, so thought I'd bring it up to do here. Also useful for quickly tracking changes to `.ma` and `.mi` files._

**Comment by smwoodman** [comment]
Yes, this is awesome thanks Jacob!

**Comment by smwoodman** [comment]
> When a glider is deployed, should we create an "issue" for that deployment? Assign pilots, ask for help, discuss, etc.? Then we could have a "roadmap" project view of our glider deployments.

@acossio started a 'Checklist for tasks associated with each ESD glider deployment' template [here](https://github.com/us-amlr/glider-lab/blob/main/.github/ISSUE_TEMPLATE/glider-deployment.md); this may be a bit different from what you're talking about though? 

Totally agreed we should have as many templates as possible (useful), e.g. I'll be making one for 'data processing steps'. And then being sure to reference them in/on the glider lab manual website so we remember to use them!

**Reply by prtda** [reply]
Ok cool! What I was thinking is different, maybe complimentary. 
Glider in the water? ---> new issue
Glider out of the water? ---> issue solved

What's the utility of this? Creating a view that is a one-stop-shop for everyone to see what is currently in the water. And I think the "roadmap" type view can give a cool timeline of what all has been deployed and when. I will work up a mockup in another project to see how this would look.... 

... so ignore this for now :)

**Comment by smwoodman** [comment]
> Column for location of task, especially for lab work. Right now we have a Location column, maybe we should change to Lab or something like that. Drop-down options: La Jolla, Humboldt, n/a
Make a project view for supervisors, based on what they want to see. Make it as easy as possible to check progress for someone who is not necessarily immersed in the day-to-day.
In general, I think we should all play with "project views" in this experimental phase.

Totally agreed all around! 

I think Tony is planning to set up a meeting with the whole team (in a few weeks?) to show the project/lab manual to GW/Heidi/Christian, so that could be a good time to discuss what they want to see in such a view.

Sidebar, I'm also starting to explore how to 'best' manage my subscriptions/notifications so that I will see a reasonable amount of things and thus will actually read them. E.g. [here](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/managing-subscriptions-for-activity-on-github/managing-your-subscriptions)

**Comment by acossio** [comment]
I think these are great ideas Jacob. Thank you for starting this discussion. 

> _Starting a discussion to list out some possible ways we might try experimenting with github issues and project features..._
> 
> Since we're all sort of learning this system together, I thought it might be good to have a place to brainstorm some ways to make the most of GitHub.
> 
> Though, these ideas may be more cumbersome than helpful, so of course take it or leave it as needed. Also, caveat, I'm new here :)
> 
> #### Project
> * Column for location of task, especially for lab work. Right now we have a `Location` column, maybe we should change to `Lab` or something like that. Drop-down options: `La Jolla`, `Humboldt`, `n/a`

Currently, we have Location with a few drop down options. La Jolla, Humboldt, n/a, and deployed. I guess we could put the service locations as well in there, ie Teledyne, for when items are away for service. I think if we title it "Lab" that might limit it more. Those are my thoughts. 
> * Make a project view for supervisors, based on what they want to see. Make it as easy as possible to check progress for someone who is not necessarily immersed in the day-to-day.
I love it. If you have time, please play around with a tab that you think might be the essentials that someone like Christian, Heidi or Eric might want to see. 

> * In general, I think we should all play with "project views" in this experimental phase.

Totally agreed. We can keep adding tabs that maybe just for Jacob, or just for Tony. 

> * When a glider is deployed, should we create an "issue" for that deployment? Assign pilots, ask for help, discuss, etc.? Then we could have a "roadmap" project view of our glider deployments.
> 
I just saw your reply on how you are thinking of making this work with the Glider in the water issue. Yes, play around with it and let us know. 

> #### Repository
> * Have a `deployment` folder in the repo with subfolders that contain the piloting files for a particular deployment. Pilots would "pull" these to their local machines to use in SFMC and push back to the repo when making any changes after sending to the glider. _Maybe this is overkill, as SFMC archives everything already. But it was something I wanted to do on my personal repo when piloting, because I use different computers depending on where I am, so thought I'd bring it up to do here. Also useful for quickly tracking changes to `.ma` and `.mi` files._

This is an interesting idea. This sounds like something that the pilots should all get together and talk about. Currently, we put all the initially loaded files on the SFMC as well as in a Google Drive folder. If I understand the second part of this, is that you want to have pilots push up when they change files on the SFMC to Github. Personally, I prefer getting the files straight from the SFMC so I know that there isn't a lapse in changes to files if I or the previous pilot didn't push changes to Github. But I think it is worth a discussion with other pilots as well.

**Reply by prtda** [reply]
> Currently, we have Location with a few drop down options. La Jolla, Humboldt, n/a, and deployed. I guess we could put the service locations as well in there, ie Teledyne, for when items are away for service. I think if we title it "Lab" that might limit it more. Those are my thoughts.

Maybe `Service Location` or similar would be a good column title then.

> This sounds like something that the pilots should all get together and talk about.

Ok I definitely need to use the established piloting workflow in practice myself before I can contribute meaningfully to a discussion. so, we'll circle back to this someday if needed!
